tests/app/main/test_suppliers.py

```diff
@@ -1665,3 +1666,192 @@ def test_should_show_email_address_even_when_refreshed(self):
 
             assert res.status_code == 200
             assert 'An email has been sent to my-email@example.com' in res.get_data(as_text=True)
+
+
+class TestJoinOpenFrameworkNotificationMailingList(BaseApplicationTest):
+    @staticmethod
+    def _common_page_asserts_and_get_form(doc):
+        assert tuple(h1.xpath("normalize-space(string())") for h1 in doc.xpath("//h1")) == (
+            "Sign up for Digital Marketplace email alerts",
+        )
+
+        form = next(
+            form for form in doc.xpath("//form[@method='POST']")
+            if urlparse(form.action)[2:] == ("/suppliers/mailing-list", "", "", "",) and
+            form.xpath(".//input[@name='email_address']")
+        )
+        assert form.xpath(".//input[@name='csrf_token']")
+        assert form.xpath(".//input[@type='submit'][@value='Subscribe']")
+
+        return form
+
+    @mock.patch("app.main.views.suppliers.DMMailChimpClient")
+    def test_get(self, mailchimp_client_class):
+        mailchimp_client_instance = mock.Mock(spec=("subscribe_new_email_to_list",))
+        mailchimp_client_instance.subscribe_new_email_to_list.side_effect = AssertionError("This should not be called")
+
+        mailchimp_client_class.side_effect = assert_args_and_return(
+            mailchimp_client_instance,
+            "not_a_real_username",
+            "not_a_real_key",
+            mock.ANY,
+        )
+
+        with self.client as c:
```

Comment:
What benefit does this `with` give us here?  `c` is only used once in the following block, is there a problem with just doing `response = self.client.get("/suppliers/mailing-list")` as we have in most other tests?

Reply:
Oh hell I think I'm supposed to be using `c.session_transaction()`, and the `c` is supposed to be the context that binds the http session together.